tools/ReleaseNotes/data/whats-new-13.md

```diff
@@ -0,0 +1,1561 @@
+var api = builder.AddUvicornApp("api", "./api", "main:app");
+
+// Extract files FROM the frontend container and copy TO the api container
+api.PublishWithContainerFiles(frontend, "./static");
+```
+
+**How it works:**
+
+1. The `frontend` resource builds inside its container, producing output files
+2. Aspire extracts those files from the frontend container
+3. The files are copied into the `api` container at `./static`
+4. The final `api` container contains both the API code and the frontend static files
+
+**Example: Serving frontend from backend**
+
+The FastAPI app can serve the static files:
+
+```python
+from fastapi import FastAPI
+from fastapi.staticfiles import StaticFiles
+
+app = FastAPI()
+
+# Serve the frontend static files
+app.mount("/", StaticFiles(directory="static", html=True), name="static")
+
+# API endpoints
+@app.get("/api/data")
+def get_data():
+    return {"message": "Hello from API"}
+```
+
+This pattern works with any resource types (.NET, Python, JavaScript) and integrates seamlessly with `aspire do` for dependency tracking, parallel execution, and incremental builds.
+
+### Dockerfile Builder API (Experimental)
+
+Aspire 13.0 introduces an experimental programmatic Dockerfile generation API that allows you to define Dockerfiles using C# code with a composable, type-safe API.
+
+> [!IMPORTANT]
+> 🧪 **Experimental Feature**: The Dockerfile Builder API is experimental and may change before general availability.
+
+```csharp
+var app = builder.AddContainer("app", "app")
+    .PublishAsDockerFile(publish =>
+    {
+        publish.WithDockerfileBuilder("/path/to/app", context =>
+        {
+            var buildStage = context.Builder
+                .From("golang:1.23", "builder")
+                .WorkDir("/build")
+                .Copy(".", "./")
+                .Run("go build -o /app/server .");
+
+            context.Builder
+                .From("alpine:latest")
+                .CopyFrom(buildStage.StageName!, "/app/server", "/app/server")
+                .Entrypoint(["/app/server"]);
+        });
+    });
+```
+
+The API provides multi-stage builds, fluent method chaining (`WorkDir`, `Copy`, `Run`, `Env`, `User`, `Entrypoint`), comments/formatting, and BuildKit features.
+
+### Certificate Management
+
+Aspire 13.0 introduces certificate management for custom certificate authorities and developer certificate trust:
+
+```csharp
+// Add custom certificate collections
+var certs = builder.AddCertificateAuthorityCollection("custom-certs")
+    .WithCertificatesFromFile("./certs/my-ca.pem")
+    .WithCertificatesFromStore(StoreName.CertificateAuthority, StoreLocation.LocalMachine);
+
+var api = builder.AddProject<Projects.Api>("api")
+    .WithReference(certs)
+    .WithDeveloperCertificateTrust(trust: true);
+```
+
+Features include loading from PEM files or certificate stores, flexible trust scoping, customizable container paths, and automatic dev certificate trust configuration.
```

Comment:
Combine this with the `Certificate Trust Across Languages` section above as they're directly related features.